### PR TITLE
fix: skip set_selection onChange events

### PR DESCRIPTION
**Pull request by z0al**
Re-implemented the way we handle subscribing to `onChange` events. Previously, we used to skip `set_selection` events, but I mistakenly removed it [during the recent upgrade](https://github.com/contentful/field-editors/pull/1414).

This PR brings back the logic.

### CODE REVIEW

**Review by 2wce — APPROVED** [review]
LGTM

**Review by chrishelgert — APPROVED** [review]
LGTM 👍

File: packages/rich-text/src/SyncEditorChanges.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      return;
+    }
+
+    lastIncomingValue.current = incomingValue;
+    setEditorValue(editor, incomingValue);
+  }, [editor, incomingValue]);
+};
+
+/**
+ * Attaches a custom onChange callback that
```
Comment (YvesRijckaert) [inline packages/rich-text/src/SyncEditorChanges.tsx:33]: This comment might not be complete? 😄

Comment (z0al) [reply packages/rich-text/src/SyncEditorChanges.tsx:33]: 🤦🏻‍♂️ I just noticed that hahah

File: packages/rich-text/src/SyncEditorChanges.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        });
+
+        if (operations.length === 0) {
+          return;
+        }
+
+        cb(document);
+      },
+    });
+  }, [editor, onChange, setEditorOnChange]);
```
Comment (YvesRijckaert) [inline packages/rich-text/src/SyncEditorChanges.tsx:56]: Does createOnChangeCallback need to be part of the dependencies here?

File: packages/rich-text/src/SyncEditorChanges.tsx (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/SyncEditorChanges.tsx:33]: chatGPT failed me :) . damn it

File: packages/rich-text/src/SyncEditorChanges.tsx (hunk 2 above)
Comment (z0al) [reply packages/rich-text/src/SyncEditorChanges.tsx:56]: No, this is an external dependency that doesn't change.

**Pull request merged**